### ISSUE TITLE
[A11y_Bug]: Getting H43 for complex table with `scope='rowgroup'`

### ISSUE THREAD

**Issue by muhamed-codecta** [IBMa/equal-access#2413]
### Project

accessibility-checker-engine

### Browser

Chrome

### Automated testing tool and ruleset.

Chrome extension

### Description

Trying to make complex table WCAG 2.2. compatible but scanner is producing violations : "Complex table does not have headers for each cell properly defined with 'header' or 'scope'" . I've created sample table that reproduces issues.

### Steps to reproduce

Run chrome scanner extension on file [complex-table-wcag.html](https://github.com/user-attachments/files/23125372/complex-table-wcag.html)

**Comment by tombrunet** [comment]
Thank you for the test case. At first glance, yes, this looks like a false positive since it is itself a heading.

**Comment by philljenkins** [comment]
Triaged,
This case applies to first column and second row. but not a high priority false positive.
Also check for other possible cases, such as `summary`
